lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -447,14 +458,34 @@ GetNominal(swift::Demangle::Demangler &dem, swift::Demangle::NodePointer node) {
   case Node::Kind::ProtocolListWithAnyObject:
   case Node::Kind::TypeAlias: {
     if (node->getNumChildren() != 2)
-      return {};
-    auto *m = node->getChild(0);
-    if (!m || m->getKind() != Node::Kind::Module || !m->hasText())
-      return {};
-    auto *n = node->getChild(1);
-    if (!n || n->getKind() != Node::Kind::Identifier || !n->hasText())
-      return {};
-    return {{m->getText(), n->getText()}};
+      return false;
+    auto *first = node->getChild(0);
+    if (!first)
+      return false;
+    if (first->getKind() == Node::Kind::Module) {
+      assert(first->hasText() && "Module node should have text!");
```

Comment:
this also needs to be a dynamic check.

Reply:
What should the helper do though? There aren't many assumptions you can make about the type's children, as both nodes could be many different things.

The first can be the module, another type, I believe also a function.

The second one can be an identifier, a private decl, or a local decl.

Reply:
I guess we could add a generic
```
[declcontext, type] = extractBinaryNode(node)
```
facility. If you don't think that's worth it, then not.

Reply:
I added two helpers for nodes that I thought made sense, but left the rest as is